files/setup-02-proxy.sh

```diff
@@ -23,10 +23,16 @@ if [ -n "${HTTP_PROXY}" ] || [ -n "${HTTPS_PROXY}" ]; then
     fi
 
     if [ ! -z "${HTTP_PROXY}" ]; then
+        HTTP_PROXY_PREFIX="http"
+        a=(`printf '%s\n' "${HTTP_PROXY//\:\/\//$'\n'}"`)
+        if [ ${#a[*]} -eq 2 ]; then
+            HTTP_PROXY_PREFIX=${a[0]}
+            HTTP_PROXY=${a[1]}
+        fi
```

Comment:
In speaking with the team about this PR, I think it makes sense to NOT to assume whether it's HTTP(s) and let customers decide. This means this logic isn't needed and I'm suggesting that we change the original behavior from specifying <proxy>:<port> to the full URL of your proxy server whether that's HTTP or HTTPS.

Reply:
Makes sense, however I think we still need to parse the proxy URL to remove the prefix so that credentials can be inserted if required?We could remove the conditional, but it might be better to add an else clause to log a failure (i.e. the user has entered a malformed URL)? I'll commit a change along these lines for you to review.

Reply:
Not sure I follow the credentials comment? `HTTP_PROXY` and `HTTPS_PROXY` is only expected to have something like "http://proxy:3128" or "https://proxy:3128". Its not expected to contain any credentials. 

Proxy credentials are captured in `PROXY_USERNAME` and `PROXY_PASSWORD` and if those variables are detected, then you'd concat the final URI string which is then passed into the respective files

Reply:
Apologies, perhaps I could have made that clearer! 

In the case of credentials being supplied, if we don't extract the users chosen protocol (`http` or `https`) from the passed `HTTP_PROXY` and `HTTPS_PROXY` variables, then we don't know what protocol to use when doing the concat of the final URI. 
Line 33 in my commit shows the issue: `HTTP_PROXY_URL="${HTTP_PROXY_PREFIX}://${PROXY_USERNAME}:${PROXY_PASSWORD}@${HTTP_PROXY}"`

Unless I'm mistaken, we need to split the `HTTP_PROXY` URL on the `://` delimiter so that we can assemble the whole URI. Admittedly we only need to do this _if_ credentials are passed. However as suggested above we could take the opportunity here to present an error in the case of a malformed URL being given by the user.

Reply:
Ah, I see what you mean now. Yes, you're correct and we can certainly throw an error if http(s) isn't detected. Your suggestion makes perfect sense and if you wan to update the implementing, we should be good!